src/mesh/impls/bout/boutmesh.cxx

```diff
@@ -975,6 +989,53 @@ void BoutMesh::createYBoundaries() {
   }
 }
 
+std::set<std::string> BoutMesh::getPossibleBoundaries() const {
+  // Result set: set so it automatically takes care of duplicates
+  std::set<std::string> all_boundaries{};
+
+  // Lambda that modifies `all_boundaries`
+  const auto get_boundaries_on_different_rank = [mesh = this, &all_boundaries](
+                                                    int x_rank, int y_rank) {
+    // Don't try to check boundaries on unphysical processors
+    if (x_rank < 0 or x_rank >= mesh->NXPE) {
+      return;
+    }
+    if (y_rank < 0 or y_rank >= mesh->NYPE) {
+      return;
+    }
+
+    // Make a copy of this mesh, EXCEPT we change the (X, Y) rank of the processor
+    BoutMesh mesh_copy{mesh->GlobalNx, mesh->GlobalNyNoBoundaries, mesh->GlobalNz,
+                       mesh->MXG, mesh->MYG, mesh->NXPE, mesh->NYPE, x_rank, y_rank,
+                       mesh->symmetricGlobalX, mesh->symmetricGlobalY, mesh->periodicX,
+                       mesh->ixseps1, mesh->ixseps2, mesh->jyseps1_1, mesh->jyseps2_1,
+                       mesh->jyseps1_2, mesh->jyseps2_2, mesh->ny_inner};
+    // We need to create the boundaries
+    mesh_copy.createXBoundaries();
+    mesh_copy.createYBoundaries();
+
+    // Get the boundaries and shove their names into the set
+    auto boundaries = mesh_copy.getBoundaries();
+    std::transform(boundaries.begin(), boundaries.end(),
+                   std::inserter(all_boundaries, all_boundaries.begin()),
+                   [](BoundaryRegionBase* boundary) { return boundary->label; });
+  };
+
+  // This is sufficient to get the SOL boundary, if it exists
+  get_boundaries_on_different_rank(NXPE - 1, 0);
+
+  // Now we need to work out which Y rank at each of the possible
+  // targets and branch cuts. This makes sure we get a processor on
+  // each possible boundary. We only need to do this at PE_XIND==0, as
+  // the other X boundary is covered by the case above
+  for (const auto& y_index :
+       {0, jyseps1_1, jyseps1_2, ny_inner - 1, ny_inner, jyseps2_1, jyseps2_2, ny - 1}) {
```

Comment:
Is it necessary to do `jyseps*` here? I think the targets (where we actually apply y-boundary conditions) are only at `0`, `ny-1`, and (if `jyseps2_1==jyseps1_2`) `ny_inner-1` and `ny_inner`.